lua/lean/infoview.lua

```diff
@@ -139,9 +174,8 @@ local paused_txt = "[PAUSED]"
 
 --- Update this info's physical contents.
 function Info:render()
+  self.div = html.Div:new()
```

Comment:
I don't want sprinklings of HTML concepts throughout the codebase -- vim is not an HTML-based editor. We of course as part of this are translating from HTML to vim-native concepts as part of rendering widgets, but that should be limited just to the interface between widgets and the rest of the plugin (i.e. somewhere should be a component that renders HTML to vim APIs, but then nowhere else is there something HTML looking).

Reply:
If you look at lean/html.lua, you'll find that this is just a way for mapping from infoview cursor location to the relevant stack of objects. This is useful beyond widgets for jump-to-location, copy-to-comment, etc.

Reply:
Right -- we should have our own native way of doing so, and map the HTML we get from widgets into it, rather than vice versa (trying to map vim things into HTML even when we're not dealing with HTML handed to us).

Reply:
Yep that's the plan, I don't intend to go any further than what we currently have at lean/html.lua.

Reply:
What I mean is nothing in lean.infoview (or its components) should be importing or talking about HTML -- the widget handler should be the only thing dealing with HTML, and then passing either strings or more likely as you say some agnostic position-or-interaction-related objects around. `components.goal` should definitely not take an argument of type `html.Div` for instance.

Here in the PR you have lots of HTML objects already being passed around.

Reply:
> passing either strings or more likely as you say some agnostic position-or-interaction-related objects around.

This seems to me like it would require creating new objects, which we should avoid if we can directly use what we already have.

> `components.goal` should definitely not take an argument of type `html.Div` for instance.

Yeah an argument isn't necessary there, but it should definitely at least _return_ an `html.Div`, otherwise we'd completely lose the context of separate goals.

> Here in the PR you have lots of HTML objects already being passed around.

I used it every where I could to preserve context, also I find it much cleaner than `table.insert`ing a `lines` array.

Reply:
Perhaps from your perspective it's just a matter of renaming the `html` module to something else?

Reply:
> This seems to me like it would require creating new objects, which we should avoid if we can directly use what we already have.

We don't already have anything -- either way we are writing a simple UI framework.

>  but it should definitely at least return an html.Div, otherwise we'd completely lose the context of separate goals.

Please lay out what you're trying to do -- I don't follow what this has to do with widgets (and there are again lots of seemingly unrelated things rolled up in the code that's here already, so I can't tell what direction you're trying to take this in.).

But no, I don't think it should return one either.

> Perhaps from your perspective it's just a matter of renaming the html module to something else?

Possibly -- one way or another we'll end up with some form of TUI, so yes the code I was playing with for widgets was using similar kinds of objects than one would find in HTML of course (a `Container` of elements, a `Button`, a `SelectableList`, etc.), so if that's what you mean by rename, sure.

Reply:
> I don't follow what this has to do with widgets
Nothing to do with widgets, just wanted to fully integrate this as soon as we've introduced it.

> Please lay out what you're trying to do
One example is copy-to-comment